docs/README.md

```diff
@@ -258,3 +258,41 @@ Here's an example for a single value return:
     Returns:
         :obj:`List[int]`: A list of integers in the range [0, 1] --- 1 for a special token, 0 for a sequence token.
 ```
+
+#### Adding a new section
+
+In ReST section headers are designated as such with the help of a line of underlying characters, e.g.,:
+
+```
+Section 1
+^^^^^^^^^^^^^^^^^^
+
+Sub-section 1
+~~~~~~~~~~~~~~~~~~
+```
+
+ReST allows the use of any characters to designate different section levels, as long as they are used consistently within the same document. For details see [sections doc](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections). Because there is no standard different documents often end up using different characters for the same levels which makes it very difficult to know which character to use when creating a new section.
+
+Specifically, if when running `make docs` you get an error like:
+```
+docs/source/main_classes/trainer.rst:127:Title level inconsistent:
+```
+you picked an inconsistent character for some of the levels.
+
+But how do you know which characters you must use for an already existing level or when adding a new level?
+
+You can use this helper script:
+```
+perl -ne '/^(.)\1{100,}/ && do { $h{$1}=++$c if !$h{$1} }; END { %h = reverse %h ; print "$_ $h{$_}\n" for sort keys %h}' docs/source/main_classes/trainer.rst
+1 -
+2 ~
+3 ^
+4 =
+5 "
+```
+
+This tells you which characters have already been assigned for each level.
+
+So using this particular example's output -- if your current section's header uses `=` as its underline character, you now know you're at level 4, and if you want to add a sub-section header you know you want `"` as it'd level 5.
+
+If you needed to add yet another sub-level, then pick a character that is not used already. That is you must pick a character that is not in the output of that script.
```

Comment:
Maybe give the list of characters available here for completeness? It's:
= - ` : ' " ~ ^ _ * + # < >

Reply:
great idea! 

I didn't know it was not any character.

Added